apps/dashboard/src/components/workflow-editor/test-workflow/test-workflow-drawer.tsx

```diff
@@ -47,24 +51,6 @@ type TestWorkflowDrawerProps = {
   initialPayload?: PayloadData;
 };
 
-// Extract only the payload part from the original schema
-const buildPayloadOnlyFormSchema = () => {
-  return z.object({
-    payload: z.string().transform((str, ctx) => {
-      try {
-        return JSON.parse(str);
-      } catch {
-        ctx.addIssue({ code: 'custom', message: 'Payload must be valid JSON' });
-        return z.NEVER;
-      }
-    }),
-  });
-};
-
-type TestWorkflowFormType = {
-  payload: string; // JSON string that gets parsed
-};
-
```

Comment:
schema and form control is not needed for payload, the Editor component handles schema validation and valid JSON already